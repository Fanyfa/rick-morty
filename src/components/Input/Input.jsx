import { func, string } from 'prop-types';
import React, { Component } from 'react';
import styles from './styles.module.scss';

class Input extends Component {
  static propTypes = {
    onChange: func.isRequired,
    name: string.isRequired,
  };

  handleOnChange = event => {
    const value = event.currentTarget.value;
    this.props.onChange(value);
  };

  render() {
    return <input className={styles.input} onChange={this.handleOnChange} />;
  }
}

export default Input;
