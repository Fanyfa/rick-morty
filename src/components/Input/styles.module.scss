@import '~styles/colors';

.input {
  border-radius: 5px;
  border: 1px solid $color3;
  outline: none;
  padding-left: 5px;
}
