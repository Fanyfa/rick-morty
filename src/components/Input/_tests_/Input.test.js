import Input from 'components/Input/';
import React from 'react';
import renderer from 'react-test-renderer';

const CLASSNAME = 'Lorem Inpsum';
const ONCHANGE = () => {};

describe('<Input>', () => {
  it('renders', () => {
    const tree = renderer.create(<Input onChange={ONCHANGE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {classname}', () => {
    const tree = renderer.create(<Input className={CLASSNAME} onChange={ONCHANGE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
