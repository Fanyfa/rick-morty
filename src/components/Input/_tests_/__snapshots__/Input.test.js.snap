// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> renders 1`] = `
<input
  className="input undefined"
  onChange={[Function]}
/>
`;

exports[`<Input> when {classname} 1`] = `
<input
  className="input Lorem Inpsum"
  onChange={[Function]}
/>
`;
